src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -442,9 +441,21 @@ def resize_subvolume(self, fs_handle, **kwargs):
                     raise VolumeException(
                         -errno.ENOENT, "Subvolume '{0}' not found, create it with " \
                         "'ceph fs subvolume create' before resizing subvolumes".format(subvolname))
-                nsize, usedbytes = sv.resize_subvolume(subvolpath, newsize, noshrink)
-                ret = 0, json.dumps([{'bytes_used': usedbytes},{'bytes_quota': nsize},
-                                     {'bytes_pcent': '{0:.2f}'.format((float(usedbytes) / nsize) * 100.0)}], indent=2), ""
+
+                try:
```

Comment:
I think it's better to try to convert it to an `int()` first. If that conversion succeds, then do `sv.resize_volume` otherwise `sv.resize_infinite`. It makes the exception handling of the `try/except` more explicit.

Reply:
Do you mean as in [patch.txt](https://github.com/ceph/ceph/files/3807113/patch.txt)? 
Yes, it is possible, but it creates an unwanted return statement as shown in [patch.txt](https://github.com/ceph/ceph/files/3807113/patch.txt). So I avoided it.

Reply:
Agree with Patrick